### PR TITLE
Load the generated files

### PR DESCRIPTION
**Pull request by anand-kamble**
This PR includes the updates to the class `data_processor`.

Now the class automatically saves the filtered data. If the filtered data already exists it will load the data directly from the existing output file (treats the previous output as cache). You can also explicitly mention not to use cache and generate new filtered data by passing `no_cache=True` in the constructor of the class.

The `data_config.json` file specifies everything about how to filter the data including the output directory, format, and file name.

I have also added a logger class that logs all the events in a text file named `copa_logs.txt`. This will help debug in the future.
The log output should look like:
```
[2024-05-09 20:19:18.361627] (DataProcessor) Initializing DataProcessor in test mode. with base_path: copa/
[2024-05-09 20:19:18.362846] (DataProcessor) Config diff is: set()
[2024-05-09 20:19:18.364254] (DataProcessor) Cache is disabled. Generating new filtered file...
[2024-05-09 20:19:18.574405] (DataProcessor) Loading file: SD_FAULT_REFERENCE.csv
[2024-05-09 20:19:18.592652] (DataProcessor) Loading file: FAIL_DEFER_REF.csv
[2024-05-09 20:19:18.619406] (DataProcessor) Loading file: EVT_EVENT.csv
[2024-05-09 20:19:18.784147] (DataProcessor) Loading file: INV_LOC.csv
[2024-05-09 20:19:18.904171] (DataProcessor) Loading file: REQ_PART.csv
[2024-05-09 20:19:19.088790] (DataProcessor) Loading file: SCHED_STASK.csv
[2024-05-09 20:19:19.220233] (DataProcessor) Loading file: REF_FAIL_SEV.csv
[2024-05-09 20:19:19.224814] (DataProcessor) Loading file: FL_LEG.csv
[2024-05-09 20:19:19.393409] (DataProcessor) Filtering file: SD_FAULT_REFERENCE.csv
[2024-05-09 20:19:19.395558] (DataProcessor) Filtering file: FAIL_DEFER_REF.csv
[2024-05-09 20:19:19.398025] (DataProcessor) Filtering file: EVT_EVENT.csv
[2024-05-09 20:19:19.401639] (DataProcessor) Filtering file: INV_LOC.csv
[2024-05-09 20:19:19.408463] (DataProcessor) Filtering file: REQ_PART.csv
[2024-05-09 20:19:19.424384] (DataProcessor) Filtering file: SCHED_STASK.csv
[2024-05-09 20:19:19.439980] (DataProcessor) Filtering file: REF_FAIL_SEV.csv
[2024-05-09 20:19:19.459012] (DataProcessor) Filtering file: FL_LEG.csv
[2024-05-09 20:19:19.558200] (DataProcessor) output directory does not exist. Creating one...
[2024-05-09 20:19:19.763637] (DataProcessor) Filtered data saved successfully.
```

With this class, you will always have filtered data to work within the project. 

Here is the demo video of how it is working on my machine.

https://github.com/anand-kamble/data-filtering/assets/78261099/abacbb49-0c5e-4f93-8379-69672c0dd7c0

### CODE REVIEW

File: src/copa_modules/_data_processor/data_processor.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        """
+        Initializes the DataProcessor object with the provided configuration and options.
+
+        Args:
+            config (DataConfig | None): The configuration object containing parameters for data processing.
+                                        If None, a CopaError will be raised.
+            base_path (str, optional): The base path where the data is located. Defaults to an empty string,
+                                        which means the data is located in the current directory.
+            test_mode (bool, optional): If True, the DataProcessor will run in test mode. Defaults to False.
+            test_rows (int, optional): The number of rows to process in test mode. Defaults to 100.
```
Comment (erlebach) [inline src/copa_modules/_data_processor/data_processor.py:41]: I I run in test mode with 100 rows, a cached file is created. If I rerun with a different number of rows, is the cache automatically overwritten? Or is the original cache used (in which case the number of rows will not be correct.)

File: Readme.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
 ## Use of LLM to improve plane maintenance
 
+# Data Processor
+
+
+### Parameters
+
+- `config` (`data_config` or `None`): The configuration object containing parameters for data processing. If `None`, a `CopaError` will be raised.
+- `base_path` (`str`, optional): The base path where the data is located. Defaults to an empty string, which means the data is located in the current directory.
```
Comment (erlebach) [inline Readme.md]: Shouldn't we state where the program should be executed from? Alternatively, if base_path is relative, relative to what?

**Review by erlebach — APPROVED** [review]
I added two comments. But I approve the code as is.

File: src/copa_modules/_data_processor/data_processor.py (hunk 1 above)
Comment (anand-kamble) [reply src/copa_modules/_data_processor/data_processor.py:41]: I have resolved this bug in the latest v1.0.2. Please have a look at the commit https://github.com/anand-kamble/data-filtering/pull/7/commits/fa3fcc8a07b0fc5d791078494ee540be8cd2a8dd

**Pull request merged**